### PR TITLE
Content loader upgrade

### PR DESCRIPTION
**Pull request by jakubDoka**
Pr improves the makeContent macro, its now lot more complex under the hood but it simplifies the defining process and adds feature of initializers. The macro is documented with example block of code.

### CODE REVIEW

File: meta.nim [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## macro utility
```
Comment (Anuken) [inline meta.nim]: 1. This should probably be in `futils`, I'd rather not have another small file here.
2. Aren't the check functions very similar to [expectKind](https://nim-lang.org/docs/macros.html#expectKind%2CNimNode%2CNimNodeKind), without the condition?

File: presets/content.nim [hunk 2]
```diff
@@ -32,66 +32,175 @@ macro defineContentTypes*(body: untyped): untyped =
...
-      if typeName notin usedNames:
-        result.add quote do:
-          var `listName`* {.inject.}: seq[`typet`]
-        usedNames.incl typeName
-      
-      #switch empty calls to constructors
-      if consn.kind == nnkCall:
-        consn = newNimNode(nnkObjConstr).add(ident(typeName))
-      
-      defaultValues.withValue(typeName, defs):
```
Comment (Anuken) [inline presets/content.nim:73]: I don't see a replacement for `defaultValues` anywhere; why was it removed?

Comment (jakubDoka) [reply presets/content.nim:73]: I thought that there is no need to do this any more as code you write inside new macro is fundamentally different. If you want to declare content with default values you write just name of the content to the correct section. The Constructor is inferred from parent type section. If macro documentation is not clear I can improve it.

File: meta.nim (hunk 1 above)
Comment (jakubDoka) [reply meta.nim]: afaik you cannot use custom message with `expect{Kind|Len}`

File: presets/content.nim (hunk 2 above)
Comment (Anuken) [reply presets/content.nim:73]: >  If you want to declare content with default values you write just name of the content to the correct section.

I don't understand what this means, and the examples don't show any default values being used (aside from `initializer`, which is not a replacement).

**Pull request closed without merging**